src/overlays/actors/ovl_Bg_Jya_1flift/z_bg_jya_1flift.c

```diff
@@ -21,31 +37,164 @@ const ActorInit Bg_Jya_1flift_InitVars = {
+    BgJya1flift_InitCollision(thisx, globalCtx);
+    if (Flags_GetSwitch(globalCtx, (thisx->params & 0x3F)) != 0) {
+        LINK_AGE_IN_YEARS == YEARS_ADULT ? BgJya1flift_ChangeDirection(this) : func_80892E0C(this);
+    } else {
+        func_80892DB0(this);
+    }
+    thisx->room = -1;
+    D_808930E0 = 1;
+    this->hasInitialized = 1;
+}
+
+void BgJya1flift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+
+    if (this->hasInitialized != 0) {
+        D_808930E0 = 0;
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void func_80892DB0(BgJya1flift* this) {
+    this->actionFunc = func_80892DCC;
+    this->dyna.actor.posRot.pos.y = finalPositions[0];
+}
+
+void func_80892DCC(BgJya1flift* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F))) {
+        BgJya1flift_ChangeDirection(this);
+    }
+}
+
+void func_80892E0C(BgJya1flift* this) {
+    this->actionFunc = BgJya1flift_DoNothing;
+    this->dyna.actor.posRot.pos.y = finalPositions[0];
+}
+
+void BgJya1flift_DoNothing(BgJya1flift* this, GlobalContext* globalCtx) {
+}
+
+void BgJya1flift_ChangeDirection(BgJya1flift* this) {
+    this->actionFunc = BgJya1flift_Move;
+    this->direction ^= 1;
+    this->dyna.actor.velocity.y = 0.0f;
+}
+
+void BgJya1flift_Move(BgJya1flift* this, GlobalContext* globalCtx) {
+    f32 tempVelocity;
+
+    Math_ApproxF(&this->dyna.actor.velocity.y, 6.0f, 0.4f);
+    if (this->dyna.actor.velocity.y < 1.0f) {
+        (tempVelocity = 1.0f);
+    } else {
+        (tempVelocity = this->dyna.actor.velocity.y);
+    }
+    if (fabsf(Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, (finalPositions[this->direction]), 0.5f,
+                                      tempVelocity, 1.0f)) < 0.001f) {
+        this->dyna.actor.posRot.pos.y = finalPositions[this->direction];
+        BgJya1flift_ResetMoveDelay(this);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_BOUND);
+    } else {
+        func_8002F974(&this->dyna.actor, NA_SE_EV_ELEVATOR_MOVE3 - SFX_FLAG);
+    }
+}
+
+void BgJya1flift_ResetMoveDelay(BgJya1flift* this) {
+    this->actionFunc = BgJya1flift_DelayMove;
+    this->moveDelay = 0;
+}
+
+void BgJya1flift_DelayMove(BgJya1flift* this, GlobalContext* globalCtx) {
+    this->moveDelay++;
+    if (this->moveDelay >= 21) {
+        BgJya1flift_ChangeDirection(this);
+    }
+}
+
+void BgJya1flift_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+    s32 pad;
+    s32 tempIsRiding;
+
+    // Room 0 is the first room and 6 is the room that the lift starts on
+    if (globalCtx->roomCtx.curRoom.num == 6 || globalCtx->roomCtx.curRoom.num == 0) {
+        this->actionFunc(this, globalCtx);
+        if (globalCtx) {}
+        tempIsRiding = (func_8004356C(&this->dyna) != 0) ? 1 : 0;
```

Comment:
anywhere a boolean is referenced in this file i would change it to true and false, and make the checks (!isRiding) and (isRiding) for example